### ISSUE TITLE
getTransactionByHash logs EOF error in terminal

### ISSUE THREAD

**Issue by frozeman** [ethereum/go-ethereum#962]
Sending the following for a non existing transaction:
```
{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x10522fd0474ee880359db0162d13ca7f57b0d7c8f32c0059ca14fc359fd34286"],"id":855}
```

Logs the following in the console:
```
I0513 18:28:32.306191   90997 xeth.go:329] EOF
```

**Comment by tgerring** [comment]
Result shows as null. What's the result from C++ ?

```
$ curl -v -X POST --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x10522fd0474ee880359db0162d13ca7f5b0d7c8f32c0059ca14fc359fd34286"],"id":855}' http://127.0.0.1:8545
* Rebuilt URL to: http://127.0.0.1:8545/
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8545 (#0)
> POST / HTTP/1.1
> User-Agent: curl/7.37.1
> Host: 127.0.0.1:8545
> Accept: */*
> Content-Length: 142
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 142 out of 142 bytes
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Wed, 13 May 2015 22:40:48 GMT
< Content-Length: 50
< 
{
	"id": 855,
	"jsonrpc": "2.0",
	"result": null
* Connection #0 to host 127.0.0.1 left intact
}
```

**Comment by tgerring** [comment]
Looks like this might occur when the RLP decoder fails, perhaps when the object is not found in the database?

https://github.com/ethereum/go-ethereum/blob/2c1b0ff17e020f300ed9d5a5a244f59b4febfe66/xeth/xeth.go#L321-L330

**Comment by tgerring** [comment]
Fixed in 12f82ab2ff71e1d93610a40cf206f3e3d3faa70d
